cpp/src/arrow/util/bit_util.cc

```diff
@@ -285,6 +285,8 @@ void UnalignedBitmapOp(const uint8_t* left, int64_t left_offset, const uint8_t*
   for (int64_t i = 0; i < length; ++i) {
     if (op(left_reader.IsSet(), right_reader.IsSet())) {
       writer.Set();
+    } else {
+      writer.Clear();
```

Comment:
this seems the right thing to do to. we can now implement a & b & c by doing 
 dst = a & b 
 dst = dst & c

and it doesn't required the dst buffer to be zero-filled to begin with. however, it may impact perf for the compute kernels - @xhochy

Reply:
Or we have two versions, one requires zero-filled and another one doesn't?

Reply:
i prefer your current version. but, will wait to hear from others.